### PR TITLE
sentinel: Add GraphQL layer for direct object access

### PR DESCRIPTION
**Pull request by efritz**
Adds resolver for Vulns and vuln matches.

## Test plan

It may only compile...

### CODE REVIEW

File: cmd/frontend/graphqlbackend/codeintel.graphql [hunk 1]
```diff
@@ -2269,3 +2269,173 @@ type AutoIndexJobDescription {
...
+TODO
+"""
+type Vulnerability implements Node {
+    """
+    TODO
+    """
+    id: ID!
+
+    """
+    TODO
```
Comment (Numbers88s) [inline cmd/frontend/graphqlbackend/codeintel.graphql:2283]: This is the cve number. Example "CVE-2023-1234"

File: cmd/frontend/graphqlbackend/codeintel.graphql [hunk 2]
```diff
@@ -2269,3 +2269,173 @@ type AutoIndexJobDescription {
...
+    """
+    affectedPackage: VulnerabilityAffectedPackage!
+
+    """
+    TODO
+    """
+    preciseIndex: PreciseIndex!
```
Comment (Numbers88s) [inline cmd/frontend/graphqlbackend/codeintel.graphql:2440]: Add sourceFile and affected locations (codeintel location)

File: cmd/frontend/graphqlbackend/codeintel.graphql [hunk 3]
```diff
@@ -2269,3 +2269,173 @@ type AutoIndexJobDescription {
...
+    """
+    TODO
+    """
+    vulnerability: Vulnerability!
+
+    """
+    TODO
+    """
+    affectedPackage: VulnerabilityAffectedPackage!
+
```
Comment (Numbers88s) [inline cmd/frontend/graphqlbackend/codeintel.graphql:2436]: Add currentVersion of the precise index uses

File: cmd/frontend/graphqlbackend/codeintel.graphql (hunk 2 above)
Comment (Numbers88s) [reply cmd/frontend/graphqlbackend/codeintel.graphql:2440]: We don't need sourceFile 😓

**Review by Numbers88s — APPROVED** [review]

File: cmd/frontend/graphqlbackend/codeintel.graphql (hunk 3 above)
Comment (efritz) [reply cmd/frontend/graphqlbackend/codeintel.graphql:2436]: Will have to wire this up post-merge.

File: cmd/frontend/graphqlbackend/codeintel.graphql (hunk 2 above)
Comment (efritz) [reply cmd/frontend/graphqlbackend/codeintel.graphql:2440]: Will have to wire this up post-merge.

**Pull request merged**